### PR TITLE
Add support for filtering in/our `iscrowd` entries from COCO

### PR DESCRIPTION
**Pull request by JanuszL**
- adds `include_is_crowd` option that allows filtering in/out
  annotation entries with `iscrowd=1` property
- adds a test for that

## Category:
**New feature** (*non-breaking change which adds functionality*)

## Description:
- adds `include_is_crowd` option that allows filtering in/out
  annotation entries with `iscrowd=1` property
- adds a test for that
- relates to https://github.com/NVIDIA/DALI/issues/2821

### Affected modules and functionalities:
- coco reader

### Key points relevant for the review:
- NA

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
    - test_coco.test_coco_include_crowd:
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [x] Other
    - operator description
- [ ] N/A

#### Requirements
- [x] Implements new requirements
- [ ] Affects existing requirements
- [ ] N/A

**REQ IDs**: RDCOCO.18

**JIRA TASK**: DALI-3378

### CODE REVIEW

**Comment by JanuszL** [comment]
See https://github.com/NVIDIA/DALI_extra/pull/120

**Comment by JanuszL** [comment]
!build

File: dali/operators/reader/coco_reader_op.cc [hunk 1]
```diff
@@ -102,12 +102,12 @@ If a file list is not provided, this argument is required.)code",
 
 If set to False, the bboxes are returned as [x, y, width, height].)code",
       false)
+  .AddOptionalArg("include_is_crowd",
```
Comment (mzient) [inline dali/operators/reader/coco_reader_op.cc:105]: I'd recommend either sticking to the exact name (no underscore) or removing the "is" entirely:

```suggestion
  .AddOptionalArg("include_iscrowd",
```
or
```suggestion
  .AddOptionalArg("include_crowd",
```

File: dali/operators/reader/loader/coco_loader.cc [hunk 2]
```diff
@@ -272,7 +272,8 @@ void ParseCategories(LookaheadParser &parser, std::map<int, int> &category_ids)
 
 void ParseAnnotations(LookaheadParser &parser, std::vector<Annotation> &annotations,
                       float min_size_threshold, bool ltrb,
-                      bool parse_segmentation, bool parse_rle) {
+                      bool parse_segmentation, bool parse_rle,
+                      bool include_is_crowd = true) {
```
Comment (mzient) [inline dali/operators/reader/loader/coco_loader.cc:276]: See the note on naming above.

File: dali/operators/reader/loader/coco_loader.cc [hunk 3]
```diff
@@ -395,14 +399,16 @@ void ParseJsonFile(const OpSpec &spec, std::vector<detail::ImageInfo> &image_inf
   RAPIDJSON_ASSERT(parser.PeekType() == kObjectType);
   parser.EnterObject();
   float sz_threshold = spec.GetArgument<float>("size_threshold");
+  bool include_is_crowd = spec.GetArgument<bool>("include_is_crowd");
```
Comment (mzient) [inline dali/operators/reader/loader/coco_loader.cc:402]: Adjust accordingly.

File: dali/test/python/reader/test_coco.py [hunk 4]
```diff
@@ -225,3 +223,49 @@ def check_coco_reader_alias(polygon_masks, pixelwise_masks):
...
+
+
+@params(True, False)
+def test_coco_include_crowd(include_is_crowd):
+    @pipeline_def(batch_size=1, device_id=0, num_threads=4)
+    def coco_pipe(include_is_crowd):
+        _, boxes, _, image_ids = fn.readers.coco(file_root=file_root,
+                                                 annotations_file=train_annotations,
+                                                 image_ids=True,
+                                                 include_is_crowd=include_is_crowd)
```
Comment (mzient) [inline dali/test/python/reader/test_coco.py:235]: Adjust accordingly.

File: dali/operators/reader/coco_reader_op.cc (hunk 1 above)
Comment (JanuszL) [reply dali/operators/reader/coco_reader_op.cc]: Fixed

**Review by mzient — APPROVED** [review]

**Review by banasraf — APPROVED** [review]

**Pull request merged**